cachi2/core/package_managers/gomod.py

```diff
@@ -549,6 +550,7 @@ def _resolve_gomod(
         "PATH": os.environ.get("PATH", ""),
         "GOMODCACHE": "{}/pkg/mod".format(tmp_dir),
         "GOSUMDB": "sum.golang.org",
+        "GOTOOLCHAIN": "local",
```

Comment:
We also need to add this variable to the expected data in the integration tests. This is what caused a lot of failures in the last CI run.

Reply:
I can do this by running `make generate-test-data`. This will also bring in the new stdlib packages from go 1.21.